File: Letter Combinations of a Phone Number.cpp
class Solution {
public:
    string str[10] = {"", "", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wxyz"};
    vector<string> answer;
    
    void dfs(string s, string digits, int n){
        int number = digits[n] - '0';
        
        if(digits.size() - 1 == n){
            for(int i = 0; i < str[number].size(); i++){
                answer.push_back(s + str[number][i]);
            }
            return;
        }
        
        for(int i = 0; i < str[number].size(); i++){
            string tmp = s + str[number][i];
            
            dfs(tmp, digits, n + 1);
        }
    }
    
    vector<string> letterCombinations(string digits) {
        vector<int> numbers;
        
        if(digits != ""){
            dfs("", digits, 0);
        }
        
        return answer;
    }
};

File: Letter Combinations of a Phone Number.js
var letterCombinations = function(digits) {
    const answer = [];
    const strings = ["", "", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wxyz"];
    const length = digits.length;
    
    if(!digits.length){
        return [];
    }
    
    const dfs = (current, index, size) => {
        const digit = parseInt(digits[index]);
        
        if(size === length - 1){
            for(let i = 0; i < strings[digit].length; i++){
                answer.push(current + strings[digit][i]);
            }
            
            return;
        }
        
        for(let i = 0; i < strings[digit].length; i++){
            const string = current + strings[digit][i];
            
            dfs(string, index + 1, size + 1);
        }
    }
    
    dfs("", 0, 0);
    
    return answer;
};